src/FileServiceManager/FileService/Controllers/FileController.cs

```diff
@@ -22,10 +23,18 @@ public FileController(IStorage storage)
             _storage = storage;
         }
 
-        [HttpGet("Download/{fileId}")]
+        [HttpGet("Download")]
         public async Task<FileResult> DownLoadFileAsync(Guid fileId, int companyId)
         {
-            return await _storage.DownLoadFileAsync(fileId, companyId);
+            var fileInfo = await _storage.GetFileAsync(fileId, companyId);
+
+            var memory = new MemoryStream();
+            using (var stream = new FileStream(fileInfo.FilePath, FileMode.Open))
+            {
+                await stream.CopyToAsync(memory);
+            }
+            memory.Position = 0;
```

Comment:
Do we really need this Position change ?

Reply:
I change this part by providing FileStream object direct to File(...